src/mission_ui.cpp

```diff
@@ -75,6 +76,9 @@ class mission_ui_impl : public cataimgui::window
         size_t window_height = str_height_to_pixels( TERMY ) / 2;
         size_t table_column_width = window_width / 2;
 
+        bool scroll_up_this_frame = false;
```

Comment:
Instead of booleans that are cleared after every frame just store the scroll offset instead. You can make it either pixels like `window_height` or a fraction from 0 to 1.

Reply:
There's the `cataimgui::scroll` enum along with `cataimgui::set_scroll` that should be used for this. It can also be expanded with more options than single line and full page if wanted.